### PR TITLE
Add test repositories before patching

### PR DESCRIPTION
**Pull request by dasantiago**
If the test suite have the test variables _XXX_TEST_REPO_ defined, the
repos defined on those variables will be added.
The patches required from those repos will be logged.

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -264,6 +265,21 @@ sub get_netboot_mirror {
...
+        my $repo = get_var($var);
+        next unless $repo;
+        zypper_call("--no-gpg-check ar -f '$repo' test-repo-$var");
+        zypper_call("ref -r $repo");
+        zypper_call("patches -r $repo", log => "patches-$var.log");
+        zypper_call("se -t package -r $repo", exitcode => [0, 104], log => "packages-$var.log");
+    }
+}
+
+
```
Comment (okurz) [inline lib/utils.pm]: The `lib/utils.pm` file is getting quite big. If you don't mind you could move this function into its own "lib" module with a good name and proper documentation. Compare to `lib/registration.pm` for an example. But the change looks fine functionality-wise so if this is urgently needed a QAM colleague could just accept as is.

Comment (mimi1vx) [inline lib/utils.pm]: is qam specific code -- > so lib/qam.pm will be good place

**Comment by dasantiago** [comment]
Changes done today:

*- Added  code to the zypper_patch test, to include the call to our function
*- Moved the function out of utils.pm to qam.pm

All the changes were tested locally:
zdup: http://tragicbox.suse.cz/tests/451
online: http://tragicbox.suse.cz/tests/450
offline: http://tragicbox.suse.cz/tests/449

**Comment by okurz** [comment]
Please squash commits. I think all four commits should go into one. Do you have a local verification run available? E.g. a URL link to a test server where you tested it?

**Comment by dasantiago** [comment]
Done.

Online migration: http://tragicbox.suse.cz/tests/456#step/zypper_patch/6
Offline migration: http://tragicbox.suse.cz/tests/455#step/patch_before_migration/14

**Pull request merged**